.github/workflows/test.yml

```
@@ -35,17 +33,17 @@ jobs:
 
       steps:
       - uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
+
+      - name: Install uv Set up Python
+        uses: astral-sh/setup-uv@v5
         with:
+          enable-cache: true
           python-version: ${{ matrix.python-version }}
+
       - name: Display Python version
-        run: python -c "import sys; print(sys.version)"
+        run: python -V
+
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install pytest pandas
-          pip install ".[optimizer]"
-      - name: Test with pytest
-        run: |
-          pytest
\ No newline at end of file
+          uv pip install -e ".[optimizer,dev]"
+          pytest
```

I suggest splitting installing dependencies and running `pytest` into two steps, just like in the `test_basic` job above.

```suggestion
      - name: Test with pytest
        run: pytest
```